src/parser/sparqlParser/SparqlQleverVisitor.cpp

```diff
@@ -856,6 +858,17 @@ GraphPatternOperation Visitor::visitSpatialQuery(
     parseSpatialQuery(spatialQuery, op);
   }
 
+  try {
+    // We convert the spatial query to a spatial join configuration and discard
+    // its result here to detect errors early and report them to the user with
+    // highlighting. It's only a small struct so not much is wasted.
+    spatialQuery.toSpatialJoinConfiguration();
+  } catch (const parsedQuery::MagicServiceException& magicExc) {
+    reportError(ctx, magicExc.what());
+  } catch (const parsedQuery::SpatialSearchException& magicExc) {
+    reportError(ctx, magicExc.what());
```

Comment:
You can simply use a single `catch` clause that generically catches `const std::exception&` and I will silence the sonar warning about catching more specific exceptions. We only care about the message after all.